source/game_sa/Garage.cpp

```diff
@@ -16,3 +19,61 @@ void CGarage::BuildRotatedDoorMatrix(CEntity* pEntity, float fDoorPosition)
     matrix.GetUp() = CVector(-fSin * vecForward.y, fSin * vecForward.x, fCos);
     matrix.GetRight() = CrossProduct(vecForward, matrix.GetUp());
 }
+
+// 0x449900
+void CGarage::StoreAndRemoveCarsForThisHideOut(CStoredCar* storedCars, int32 maxSlot)
+{
+    maxSlot = std::min(maxSlot, 4);
+
+    for (auto i = 0; i < 4; i++)
+        storedCars[i].Clear();
+
+    for (auto i = vehPool->GetSize(); i; i--) {
+        if (auto veh = vehPool->GetAt(i - 1)) {
+            if (IsPointInsideGarage(veh->GetPosition()) && veh->m_nCreatedBy != MISSION_VEHICLE) {
+                if (vehIdx < maxSlot && !EntityHasASphereWayOutsideGarage(veh, 1.0f)) {
+                    storedCars[vehIdx++].StoreCar(veh);
+                }
+
+                FindPlayerInfo().CancelPlayerEnteringCars(veh);
+                CWorld::Remove(veh);
+
+                delete veh;
+            }
+        }
+    }
+
+    for (i = vehIdx; i < 4; i++)
+        storedCars[i].Clear();
```

Comment:
Srry, my bad, I broke it.

```suggestion

    auto storedCarIdx{0u};
    for (auto i = vehPool->GetSize(); i; i--) {
        if (auto veh = vehPool->GetAt(i - 1)) {
            if (IsPointInsideGarage(veh->GetPosition()) && veh->m_nCreatedBy != MISSION_VEHICLE) {
                if (storedCarIdx < maxSlot && !EntityHasASphereWayOutsideGarage(veh, 1.0f)) {
                    storedCars[storedCarIdx++].StoreCar(veh);
                }

                FindPlayerInfo().CancelPlayerEnteringCars(veh);
                CWorld::Remove(veh);

                delete veh;
            }
        }
    }

    // Clear slots with no vehicles in it
    for (auto i = storedCarIdx; i < 4; i++) {
        storedCars[i].Clear();
    }
```